internal/command/console_state.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) The OpenTofu Authors
+// SPDX-License-Identifier: MPL-2.0
+// Copyright (c) 2023 HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package command
+
+import (
+	"strings"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/hclsyntax"
+)
+
+type consoleBracketState struct {
+	openNewLine int
+	brace       int
+	bracket     int
+	parentheses int
+	buffer      []string
+}
+
+// commandInOpenState return an int to inform if brackets are open
+// or if any escaped new lines
+// in the console and we should hold off on processing the commands
+// it returns 3 states:
+// -1 is returned the is an incorrect amount of brackets.
+// for example "())" has too many close brackets
+// 0 is returned if the brackets are closed.
+// for examples "()" or "" would be in a close bracket state
+// >=1 is returned for the amount of open brackets.
+// for example "({" would return 2. "({}" would return 1
+func (c *consoleBracketState) commandInOpenState() int {
```

Comment:
nit: I would prefer a new int type like `commandStatus` with a few constants to represent potential statuses like
* `commandStatusMissingBrackets`
* `commandStatusClosedBrackets`
* `commandStatusPendingBrackets`

It would be easier to read instead of int comparison.

Reply:
I did have this originally, but found it was nicer UX to match the prompt to the number of open brackets. So two open brackets appears as:

```
> {
. (
.. [
...  
```

so started using ints